packages/react/src/use-chat.ts

```diff
@@ -522,6 +526,42 @@ By default, it's set to 0, which will disable the feature.
 
       event?.preventDefault?.();
 
+      let files: MessageFile[] = [];
+
+      if (options.files) {
+        if (options.files instanceof FileList) {
+          for (const file of Array.from(options.files)) {
+            const { name } = file;
+
+            const dataUrl = await new Promise<string>((resolve, reject) => {
+              const reader = new FileReader();
+              reader.onload = readerEvent => {
+                resolve(readerEvent.target?.result as string);
+              };
+              reader.onerror = error => reject(error);
+              reader.readAsDataURL(file);
+            });
+
+            files.push({
+              type: 'data-url',
+              name,
+              dataUrl,
+            });
+          }
+        } else if (Array.isArray(options.files)) {
+          for (const file of options.files) {
+            const { name, url, type } = file;
+
+            files.push({
+              type: 'url',
+              name,
+              contentType: type || 'application/octet-stream',
+              url,
+            });
+          }
+        }
```

Comment:
should throw in case it's something else (which is unsupported)

Reply:
done!